packages/itwinui-react/src/core/DatePicker/DatePicker.tsx

```diff
@@ -226,6 +227,8 @@ export const DatePicker = React.forwardRef((props, forwardedRef) => {
     startDate,
     endDate,
     isDateDisabled,
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
rules of hooks shouldn't be disabled. move `useId()` out into a variable

Reply:
Re-enabled hook and created the variable for useId() [here](https://github.com/iTwin/iTwinUI/pull/1486/commits/9c9f1d00e7f9cc3ec499fccbb5549154a7f2c72f).